File: rb101/rb101-rb109-small_problems/easy7/multiplicative_average.rb
def show_multiplicative_average(num_array)
  num_product = num_array.inject(:*)
  average = num_product / num_array.size.to_f
  puts "The result is #{format('%.3f', average)}"
end

show_multiplicative_average([3, 5])
show_multiplicative_average([6])
show_multiplicative_average([2, 5, 7, 11, 13, 17])

# ALTERNATIVE SOLUTION:

# def show_multiplicative_average(numbers)
#   product = 1.to_f
#   numbers.each { |number| product *= number }
#   average = product / numbers.size
#   puts format('%.3f', average)
# end

# FURTHER EXPLORATION:

# If we omit the #to_f at the beginning of the method then the
# average will some out to a less-precise whole number because you'll
# be performing integer division, which truncates any numbers after a
# the decimal (it rounds down or 'floors' the result).

File: js210/js210-small_problems/easy4/multiplicativeAverage.js
"use strict";

function showMultiplicativeAverage(numbers) {
  const product = numbers.reduce((acc, currNum) => acc * currNum);
  return (product / numbers.length).toFixed(3);
}

console.log(showMultiplicativeAverage([3, 5]));                   // "7.500"
console.log(showMultiplicativeAverage([2, 5, 7, 11, 13, 17]));    // "28361.667"


// ALTERNATIVE SOLUTION:

// function showMultiplicativeAverage(numbers) {
//   let product = 1;

//   for (let i = 0; i < numbers.length; i += 1) {
//     product *= numbers[i];
//   }

//   return (product / numbers.length).toFixed(3);
// }
